### PR TITLE
AEA-1198 add more contribution guides, add PoR first draft

### LINKED ISSUE #2091: Contribution guide
**Is your feature request related to a problem? Please describe.**
Incomplete

**Describe the solution you'd like**
Add `make` commands in README.md so everyone knows how to use develop and run demos on it. (`--local`)

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR adds docs on PoR and interaction protocols, extends contribution guide.

## Fixes

#2091, #2078, #1923

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -127,7 +127,34 @@ The following dependency is **only relevant if you intend to contribute** to the
 
 The following steps are **only relevant if you intend to contribute** to the repository. They are **not required** for agent development.
 
-- To install development dependencies (here optionally skipping `Pipfile.lock` creation):
+### Recommended
```
Comment (marcofavorito) [inline README.md:130]: what about building docs?

File: README.md [hunk 2]
```diff
@@ -127,7 +127,34 @@ The following dependency is **only relevant if you intend to contribute** to the
...
+      make new_env
+      pipenv shell
+      make install_env
+
+- For linting and static analysis use:
+
+      make lint
+      make static
+      make pylint
+      make security
```
Comment (marcofavorito) [inline README.md:143]: `misc_checks`?

File: README.md [hunk 3]
```diff
@@ -127,7 +127,34 @@ The following dependency is **only relevant if you intend to contribute** to the
...
+
+- For testing aea.{SUBMODULE} with tests/test_{TESTMODULE} use:
+
+      make dir={SUBMODULE} tdir={TESTMODULE} test-sub
+
+  e.g.
+
+      make dir=cli tdir=cli test-sub
+
+- To run demos against local packages use flag `--local` in `aea` CLI commands.
```
Comment (marcofavorito) [inline README.md:153]: ... right after `add` subcommand

Comment (marcofavorito) [reply README.md:153]: or: 

> e.g. `aea add --local`

File: README.md (hunk 1 above)
Comment (marcofavorito) [reply README.md:130]: also, it seems something like `docs_serve` is missing. Maybe worth explaining how to actually serve docs so to explore them?

Comment (DavidMinarsch) [reply README.md:130]: That's mentioned further down..

File: README.md (hunk 3 above)
Comment (DavidMinarsch) [reply README.md:153]: They'll figure this out when they use it..there are many commands which accept local

File: README.md (hunk 2 above)
Comment (DavidMinarsch) [reply README.md:147]: added package checks

File: README.md [hunk 4]
```diff
@@ -127,7 +131,36 @@ The following dependency is **only relevant if you intend to contribute** to the
...
 The following steps are **only relevant if you intend to contribute** to the repository. They are **not required** for agent development.
 
-- To install development dependencies (here optionally skipping `Pipfile.lock` creation):
+### Recommended commands
+
+- The simplest way to get setup for development on the framework is to run the following:
+
+      make new_env
+      pipenv shell
+      make install_env
```
Comment (solarw) [inline README.md:140]: why not combine all three commands with one make command?

**Review by solarw — APPROVED** [review]
LGTM

**Pull request merged**